yarn-project/bot/src/amm_bot.ts

```diff
@@ -35,24 +37,34 @@ export class AmmBot extends BaseBot {
     const { feePaymentMethod } = this.config;
     const { wallet, amm, token0, token1 } = this;
 
-    this.log.verbose(`Preparing tx with ${feePaymentMethod} fee to swap tokens`, logCtx);
+    const balances = this.getBalances();
+    this.log.info(`Preparing tx with ${feePaymentMethod} fee to swap tokens. Balances: ${jsonStringify(balances)}`, {
+      ...logCtx,
+      balances,
+    });
 
-    const ammBalances = await this.getAmmBalances();
-    const amountIn = TRANSFER_AMOUNT;
+    // 1000 ± 200
+    const amountIn = Math.floor(TRANSFER_BASE_AMOUNT + (Math.random() - 0.5) * TRANSFER_VARIANCE);
     const nonce = Fr.random();
 
+    const [tokenIn, tokenOut] = Math.random() < 0.5 ? [token0, token1] : [token1, token0];
```

Comment:
Ah, this will cause the e2e_bot test to fail 50% of the time 🤦 
I'll update the test